CorinneTest/Factory.cs

```diff
@@ -0,0 +1,158 @@
+                else if (type == 5) { Person newPerson = MakeBoy(lowAge, highAge); }
+                else if (type == 6) { Person newPerson = MakeChild(lowAge, highAge); }
+                else if (type == 7) { Person newPerson = MakeSeniorWoman(lowAge, highAge); }
+                else if (type == 8) { Person newPerson = MakeSeniorMan(lowAge, highAge); }
+                else if (type == 9) { Person newPerson = MakeSeniorPerson(lowAge, highAge); }
+                list.Add(newPerson);
+            }
+                
+            return list;
+        }
+
+        public static DateTime? GenRanAdultBirthDate(int lowAge, int highAge)
+        {
+            if (highAge < 18)
+                return null;
+            DateTime start = new DateTime((2018 - highAge), 5, 29);
+            DateTime minDate = new DateTime((2018 - lowAge), 5, 29);
+            int range = (minDate - start).Days;
+            return start.AddDays(rnd.Next(range));
+        }
+
+        public static DateTime? GenRanChildBirthDate(int lowAge, int highAge)
+        {
+            if (lowAge >= 18) { return null; }
+            if (highAge < 18) { DateTime maxDate = new DateTime((2018 - highAge), 5, 29); }
+            else { DateTime maxDate = new DateTime(2000, 5, 29); }
+            int range = (DateTime.Today-maxDate).Days;
+            return DateTime.Today.AddDays(rnd.Next(range));
+        }
+
+        public static DateTime? GenRanSeniorBirthDate(int lowAge, int highAge)
+        {
+            if (highAge < 6) { return null; }
+            if (lowAge > 65) { DateTime minDate = new DateTime((2018 - lowAge), 5, 29);}
+            else { DateTime minDate = new DateTime(1953, 5, 29); }
+            DateTime start = new DateTime((2018-highAge), 5, 29);
+            int range = (minDate - start).Days;
+            return start.AddDays(rnd.Next(range));
+        }
+
+
+        public static DateTime GenRanBirthDate(int lowAge, int highAge)
+        {
+            DateTime start = new DateTime((2018-highAge), 5, 29);
+            DateTime minDate = new DateTime((2018 - lowAge), 5, 29);
+            int range = (minDate - start).Days;
+            return start.AddDays(rnd.Next(range));
+        }
+
+        public static int? GenAdultAge(int highAge)
+        {
+            if (highAge < 28)
+                return null;
+            int age = rnd.Next(18, highAge);
+            return age;
+        }
+
+        public static int? GenChildAge(int lowAge)
+        {
+            if (lowAge >= 18)
+                return null;
+            int age = rnd.Next(lowAge, 18);
+            return age;
+        }
+
+        public static int? GenSeniorAge(int highAge)
+        {
+            if (highAge < 65)
+                return null;
+            int age = rnd.Next(65, highAge);
+            return age;
+        }
+
+        public static int GenAge(int lowAge, int highAge)
+        {
+            int age = rnd.Next(lowAge, highAge);
+            return age;
+        }
+
```

Comment:
maybe make #region #endregion for similar methods (like all GenAge stuff in a region)?

Reply:
good idea - did this or GenAge, GenBirthDate, and MakePerson